scipy/optimize/cython_optimize/examples/zeros_array_examples.pyx

```diff
@@ -0,0 +1,143 @@
+"""
+Use an array to hold callback args so there's no Python and functions can free
+the global interpreter lock.
+"""
+
+from __future__ import division, print_function, absolute_import
+from math import exp, sin
+
+from .. cimport zeros_array
+
+DEF N = 6  # number of arguments
+
+NUM_OF_IRRAD = 10
+IL = [sin(il) + 6.0 for il in range(NUM_OF_IRRAD)]
+ARGS = (1e-09, 0.004, 10.0, 0.27456)
+TOL, MAXITER = 1.48e-8, 50
+XTOL, RTOL, MITR = 0.001, 0.001, 10
+
+
+# governing equations
+cdef double f_solarcell(int n, double* args):
```

Comment:
@mikofski maybe a newb question here... I'm confused about the signature here as compared to the other approaches. Why ``int n`` instead of ``double i``? In the other examples this is a function with two parameters: ``i`` and ``args``. Here, it seems that ``i`` has been absorbed into ``args`` and replaced with an unused ``n``.

array: ``cdef double f_solarcell(int n, double* args):``
tuple: ``cdef double f_solarcell(double i, tuple args):``
struct: ``cdef double f_solarcell(double i, void *args):``

maybe something to point out in ``__init__.py``?

Reply:
> IMO it's okay that the callback has a parameter that seems unused. 

Doesn't bother me if it's documented.

> int, array is the most troublesome because you can't allocate an array in gcc on the stack, so I have to pick an arbitrary size MAXARGS for the array size. If we were using clang then I could set the size to n. I'm okay with dumping this signature if no one likes it, but there was a similar version in integrate.quad

I was guessing that the array type was going to be the easiest to use. Do you recommend ``zeros_struct`` over ``zeros_array``? If so, perhaps a comment in the intro docs is appropriate.

Reply:
I added more to the documentation that I hope clears this up better?